### PR TITLE
Renamed all JavaScript lessons and updated links to them.

### LINKED ISSUE #1229: Rename all numbered JavaScript lessons to include memorable name
It's too easy to forget which lesson is which.

In the [JS folder](https://github.com/Techtonica/curriculum/tree/master/javascript)

- javascript-1 => javascript-1-variables (Heading: JavaScript 1 - Variables, Strings, Numbers)
- javascript-2 => javascript-2-arrays-functions 
- javascript-3 => javascript-3-conditionals  (heading: JavaScript 3 - Conditionals, Comparisons, Booleans)
- javascript-4 => javascript-4-loops (heading: JavaScript 4: Loops)
- javascript-5 => javascript-5-switch (heading: JavaScript 5: Switch Statements)
- javascript-9 => javascript-9-async (heading: JavaScript 9: Async, Callbacks, Promises)

Change the heading at the top of the file to match

Update all internal links in the curriculum to point to these new names and to include them in the link text, e.g. the link text should be "JavaScript 1 - **Variables**", not just "JavaScript 1"

### PR DESCRIPTION
**Pull request by ishaan2908**
Fixes #1229 

Updated and Renamed the files. Please review! @vegetabill

### CODE REVIEW

File: javascript/javascript-4-loops.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ About 2 hours
 
 ### Prerequisites
 
-- [JavaScript 3 lesson](/javascript/javascript-3.md)
+- [JavaScript 3 - Conditionals lesson](/javascript/javascript-3-conditionals.md)
```
Comment (Kundan28) [inline javascript/javascript-4-loops.md]: Make it `JavaScript 3 - Conditionals, Comparisons, Booleans lesson`

File: javascript/javascript-5-switch.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# JavaScript 5
+# JavaScript 5 - Switch
```
Comment (Kundan28) [inline javascript/javascript-5-switch.md]: Make it `Switch Statements`

File: javascript/javascript-1-variables.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ JavaScript is used on the frontend of almost every website. It's also a widely-u
 [JavaScript I (video walkthrough of slides)](https://drive.google.com/open?id=1tCb0KPmGWV2Gyylbh8pf9Pc7cD7ROixR)
 - Please watch the video the first time without working along with the demonstration. Just absorb the concepts. Then, you can watch the demonstration a second time and code with the instructor if you like.
 
-[JavaScript  I (slides](https://drive.google.com/open?id=1WIm5UCQL9TOsmW5X6suapBhyEqtk8Y2oLKb5gUqbzus)
+[JavaScript  I (slides)](https://drive.google.com/open?id=1WIm5UCQL9TOsmW5X6suapBhyEqtk8Y2oLKb5gUqbzus)
```
Comment (Kundan28) [inline javascript/javascript-1-variables.md]: These should be `JavaScript I - Variables, Strings, Numbers` too.

**Comment by ishaan2908** [comment]
I'll make the required changes! @Kundan28

**Comment by ishaan2908** [comment]
I have made the requested changes @Kundan28 . Please review them!

**Comment by alodahl** [comment]
hi @ishaan2908, thank you for linking the issue. Could you change the PR title to specifics about your changes?  To be frank, `updated files and links` could apply to most of our issues, and we mentors have a lot to juggle without having to guess or remember what this PR is about.

**Comment by ishaan2908** [comment]
Done! @alodahl . I have changed the PR title.

File: javascript/javascript-5-switch.md (hunk 2 above)
Comment (alodahl) [reply javascript/javascript-5-switch.md]: have you seen the suggested changes button, @Kundan28 ?
image

File: README.md [hunk 4]
```diff
@@ -87,11 +87,11 @@ Before beginning the program, apprentices meet weekly for 4 weeks to learn JavaS
 1. [Debugging](/debugging/debugging.md)
 
 ### Week 3 - JavaScript Fundamentals
-1. [JavaScript 1](/javascript/javascript-1.md)
-1. [JavaScript 2](/javascript/javascript-2.md)
-1. [JavaScript 3](/javascript/javascript-3.md)
-1. [JavaScript 4](/javascript/javascript-4.md)
-1. [JavaScript 5](/javascript/javascript-5.md)
+1. [JavaScript 1 - Variables, Strings, Numbers](/javascript/javascript-1-variables.md)
+1. [JavaScript 2 - Array Functions](/javascript/javascript-2-array-functions.md)
```
Comment (alodahl) [inline README.md]:
```suggestion
1. [JavaScript 2 - Arrays, Functions](/javascript/javascript-2-arrays-functions.md)
```

It's about arrays and functions, not "array functions"

Comment (alodahl) [reply README.md]: also, please change all other instances accordingly 😊

Comment (ishaan2908) [reply README.md]: On it. I'm making the changes @alodahl

**Comment by ishaan2908** [comment]
I have made all the requested changes. Please review! @alodahl @vegetabill

File: README.md [hunk 5]
```diff
@@ -87,11 +87,11 @@ Before beginning the program, apprentices meet weekly for 4 weeks to learn JavaS
 1. [Debugging](/debugging/debugging.md)
 
 ### Week 3 - JavaScript Fundamentals
-1. [JavaScript 1](/javascript/javascript-1.md)
-1. [JavaScript 2](/javascript/javascript-2.md)
-1. [JavaScript 3](/javascript/javascript-3.md)
-1. [JavaScript 4](/javascript/javascript-4.md)
-1. [JavaScript 5](/javascript/javascript-5.md)
+1. [JavaScript 1 - Variables, Strings, Numbers](/javascript/javascript-1-variables.md)
+1. [JavaScript 2 - Arrays, Functions](/javascript/javascript-2-array-functions.md)
```
Comment (alodahl) [inline README.md]: thank you.. please change the link everywhere as well
```suggestion
1. [JavaScript 2 - Arrays, Functions](/javascript/javascript-2-arrays-functions.md)
```

**Comment by ishaan2908** [comment]
I have added changes to the parts I missed and updated rest of the files @alodahl

**Comment by vegetabill** [comment]
Thanks for the revisions. Almost complete... Using @alodahl 's suggested search query, these are still not fixed:

recursion/recursion.md:10:- [JavaScript 1 lesson](../javascript/javascript-1.md)
recursion/recursion.md:11:- [JavaScript 2 lesson](../javascript/javascript-2.md)
recursion/recursion.md:12:- [JavaScript 3 lesson](../javascript/javascript-3.md)
recursion/recursion.md:13:- [JavaScript 4 lesson](../javascript/javascript-4.md)

api/apis-and-json.md:17:- [Promises and Callbacks](/javascript/javascript-9.md)

**Comment by ishaan2908** [comment]
My bad, I overlooked these. I have made the changes now! @vegetabill

**Comment by vegetabill** [comment]
@ishaan2908 thanks looks good!

**Comment by Kundan28** [comment]
Updated in scoresheet too.

**Pull request merged**